activerecord/lib/active_record/associations/singular_association.rb

```diff
@@ -31,24 +31,17 @@ def force_reload_reader
       end
 
       private
+
+        def target_scope
+          super.limit!(1)
+        end
+
         def scope_for_create
           super.except!(klass.primary_key)
         end
 
         def find_target
-          scope = self.scope
-          return scope.take if skip_statement_cache?(scope)
```

Comment:
I've reverted this PR in 93c94973ab574083a8cd1868aae03993ca156c34 since `scope.take` is not the same with `scope.to_a.first`.

Reply:
Can you add a test so the regression doesn't happen again?

Reply:
Sure, I'll do that later.

Reply:
I can't see how `scope.take` can be different from `scope.to_a.first`  for the `scope` with `limit(1)`. If can explain it here, I can make the regression test myself and save you some time.

Reply:
I've already added the regression test ba4e68f577efc76f351d30a2914e29942b97830e.